src/lib/salad/rope.h

```diff
@@ -440,74 +456,92 @@ rope_extract(struct rope *rope, rope_size_t offset)
 }
 
 /**
- * Erase a single element from the rope. This is a straightforward
- * implementation for a single-element deletion from a rope. A
- * generic cut from a rope involves 2 tree splits and one merge.
+ * Erase `*p_size` elements from the rope if it fits into single node.
+ * Otherwise erase as many elements as fits into the node which holds data
+ * at given offset. Upon return `p_size` is decreased to the number of
+ * deleted elements.
  *
- * When deleting a single element, 3 cases are possible:
- * - offset falls at a node with a single element. In this case we
- *   perform a normal AVL tree delete.
- * - offset falls at the end or the beginning of an existing node
- *   with leaf_size > 1. In that case we trim the existing node
- *   and return.
- * - offset falls inside an existing node. In that case we split
+ * When deleting, 3 cases are possible:
+ * - offset is at the beginning of the node and number of deleted elements
+ *   is more then or equal to the number elements in the node. In this
```

Comment:
You seem to misuse `then` where `than` is needed. Please, check all the usages and fix, in the commit message too.

Reply:
Fixed.